Microsoft.Toolkit.Uwp.UI/Extensions/Markup/OnDevice.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.ApplicationModel;
+using Windows.UI.Xaml.Markup;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions.Markup
+{
+    /// <summary>
+    /// The OnDevice markup extension allows you to customize UI appearance on a per-DeviceFamily basis.
+    /// </summary>
+    /// <example>
+    /// The next TextBlock will show the text 'Hello' on desktop computers, 'World' on Xbox and 'Hi' on all other devices
+    /// <code>
+    ///     <TextBlock Text="{helpers:OnDevice Default=Hi, Desktop=Hello, Xbox=World}"
+    ///                xmlns:helpers="using:Microsoft.Toolkit.Uwp.UI.Extensions.Markup" />
+    /// </code>
+    /// </example>
+    public class OnDevice : MarkupExtension
+    {
+        /// <summary>
+        /// Gets or sets the default value for this property
+        /// </summary>
+        public object Default { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value for this property on Desktop
+        /// </summary>
+        public object Desktop { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value for this property on Holographic
+        /// </summary>
+        public object Holographic { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value for this property on IoT
+        /// </summary>
+        public object IoT { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value for this property on Team
+        /// </summary>
+        public object Team { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value for this property on Xbox
+        /// </summary>
+        public object Xbox { get; set; }
+
+        /// <summary>
+        /// Returns an object value for the current DeviceFamily, Default is used when it is not set.
+        /// </summary>
+        /// <returns>The object value to set on the property where the extension is applied.</returns>
+        protected override object ProvideValue()
+        {
+            string deviceFamily = null;
+            if (DesignMode.DesignMode2Enabled)
+            {
+                // TODO: detect DeviceFamily in XAML Designer (device dropdown)
+                // deviceFamily = ???
+            }
+            else
+            {
+                deviceFamily = Windows.System.Profile.AnalyticsInfo.VersionInfo.DeviceFamily;
```

Comment:
Use our [AnalyticsInfo extension](https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/master/Microsoft.Toolkit.Uwp.SampleApp/Common/AnalyticsVersionInfoExtensions.cs#L20) method instead of duplicating this logic here with strings, then you can use the enum returned.

Reply:
The AnalyticsInfo extension also supports a Tablet value. Should OnDevice have a Tablet property? 

Why is a tablet different than a Desktop, only the Touch versus Mouse. What if I connect a mouse to my tablet. I prefer to keep it as it is.

Reply:
The AnaliticsInfo extension is part of the SampleApp and must be moved to Microsoft.Toolkit befor it can be used. Do we do this?

Reply:
Ah, I forgot it was only part of the sample, I think it'd make sense to move as an Extension in the `Microsoft.Toolkit.Uwp` namespace.

The tablet mode is a specific mode you can put windows in:
![image](https://user-images.githubusercontent.com/24302614/48923086-5da27880-ee60-11e8-8f16-7ad1e093ff03.png)
I think it'd be useful to have `Tablet` as an option.

Reply:
Great, I will do that. Thanks

Reply:
I have second thoughts about a Tablet property. See my code below. The Text of The Textblock would be Hello when running on Desktop and World when running on Tablet, otherwise it is Hi (Xbox, Holo, IoT, etc). What happens when the user is running the app and then switches from Desktop to Tablet mode. Would the Text have to change on the fly? If yes, how do we implement this?

```
<TextBlock Text="{helpers:OnDevice Default=Hi, Desktop=Hello, Tablet=World}"
           xmlns:helpers="using:Microsoft.Toolkit.Uwp.UI.Extensions.Markup" />
```